src/components/shared/VirtualList.js

```diff
@@ -141,59 +159,83 @@ class VirtualListInner extends PureComponent {
   }
 }
 
-VirtualListInner.propTypes = {
-  itemHeight: PropTypes.number.isRequired,
-  className: PropTypes.string,
-  renderItem: PropTypes.func.isRequired,
-  items: PropTypes.array.isRequired,
-  specialItems: PropTypes.array.isRequired,
-  visibleRangeStart: PropTypes.number.isRequired,
-  visibleRangeEnd: PropTypes.number.isRequired,
-  columnIndex: PropTypes.number.isRequired,
+type VirtualListProps = {|
+  +itemHeight: CssPixels,
+  +className: string,
+  +renderItem: RenderItem,
+  +items: *[],
+  +focusable: boolean,
+  +specialItems: *[],
+  +onKeyDown: KeyboardEvent => void,
+  +onCopy: Event => void,
+  +disableOverscan: boolean,
+  +columnCount: number,
+|};
+
+type Geometry = {
+  // getBoundingClientRect in the Flow definitions is wrong, and labels the return values
+  // as a ClientRect, and not a DOMRect. https://github.com/facebook/flow/issues/5475
+  //
+  // Account for that here:
+  outerRect: DOMRect | ClientRect,
+  innerRectY: CssPixels,
 };
 
-class VirtualList extends PureComponent {
-  constructor(props) {
-    super(props);
-    this._onScroll = this._onScroll.bind(this);
-    this._onCopy = this._onCopy.bind(this);
-    this._geometry = undefined;
-    this._containerCreated = elem => {
-      this._container = elem;
-    };
-    this._innerCreated = elem => {
-      this._inner = elem;
-    };
-  }
+class VirtualList extends React.PureComponent<VirtualListProps> {
+  _container: ?HTMLDivElement;
+  _inner: ?VirtualListInner;
+  _geometry: ?Geometry;
+
+  _takeContainerRef = (element: ?HTMLDivElement) => {
+    this._container = element;
+  };
+
+  _innerCreated = (element: ?VirtualListInner) => {
+    this._inner = element;
+  };
 
   componentDidMount() {
     document.addEventListener('copy', this._onCopy, false);
-    this._container.addEventListener('scroll', this._onScroll);
+    const container = this._container;
+    if (!container) {
+      throw new Error(
+        'The container was assumed to exist while mounting The VirtualList.'
+      );
+    }
+    container.addEventListener('scroll', this._onScroll);
     this._onScroll(); // for initial size
   }
 
   componentWillUnmount() {
     document.removeEventListener('copy', this._onCopy, false);
-    this._container.removeEventListener('scroll', this._onScroll);
+    const container = this._container;
+    if (!container) {
+      throw new Error(
+        'The container was assumed to exist while unmounting The VirtualList.'
+      );
+    }
+    container.removeEventListener('scroll', this._onScroll);
   }
 
-  _onScroll() {
+  _onScroll = () => {
     this._geometry = this._queryGeometry();
     this.forceUpdate();
-  }
+  };
 
-  _onCopy(event) {
+  _onCopy = (event: Event) => {
```

Comment:
same thing about SyntheticClipboardEvent here, although using `*` could be good enough here (we don't really care about the type in this function, merely passing it to the parent)